rebuild/src/addon.h

```diff
@@ -0,0 +1,177 @@
+#ifndef BLST_TS_ADDON_H__
+#define BLST_TS_ADDON_H__
+
+#include <openssl/rand.h>
+
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <string_view>
+
+#include "blst.hpp"
+#include "napi.h"
+
+// TODO: these should come out post PR review
+using std::cout;
+using std::endl;
+
+#define BLST_TS_RANDOM_BYTES_LENGTH 8U
+
+#define BLST_TS_FUNCTION_PREAMBLE(info, env, module)                           \
+    Napi::Env env = info.Env();                                                \
+    Napi::EscapableHandleScope scope(env);                                     \
+    BlstTsAddon *module = env.GetInstanceData<BlstTsAddon>();
+
+#define BLST_TS_IS_INFINITY                                                \
+    Napi::Env env = info.Env();                                                \
+    Napi::EscapableHandleScope scope(env);                                     \
+    return scope.Escape(Napi::Boolean::New(env, _point->IsInfinite()));
+
+#define BLST_TS_SERIALIZE_POINT(macro_name)                                \
+    Napi::Env env = info.Env();                                                \
+    Napi::EscapableHandleScope scope(env);                                     \
+    bool compressed{true};                                                     \
+    if (!info[0].IsUndefined()) {                                              \
+        compressed = info[0].ToBoolean().Value();                              \
+    }                                                                          \
+    Napi::Buffer<uint8_t> serialized = Napi::Buffer<uint8_t>::New(             \
+        env,                                                                   \
+        compressed ? BLST_TS_##macro_name##_LENGTH_COMPRESSED                  \
+                   : BLST_TS_##macro_name##_LENGTH_UNCOMPRESSED);              \
+    _point->Serialize(compressed, serialized.Data());                          \
```

Comment:
can `_point` be null or invalid?

Reply:
address in response here:
https://github.com/ChainSafe/blst-ts/pull/119#discussion_r1517944776